### DISCUSSION TITLE
Federation link on restart

### DISCUSSION THREAD

**Discussion by cbcercas** [rabbitmq/rabbitmq-server#3979] [Questions]
Hello,
I created a federation with 2 cluster and federated queues
all working fine until I restart the downstream cluster, I have to update every policy to link show up again.

Here the upstream config
Name: Federation Project_1
URI: amqp://project_1_admin:[redacted]@mqproject_1

And the policy :
Name : Policy project_1
Pattern : ^project_1\.
Apply to : queues
Definition : federation-upstream: Federation Project_1
Priority :10

Thanks for your help.

**Comment by lukebakken** [comment]
We need to know the following:
* Cluster size
* RabbitMQ version
* Erlang version
* Operating system and version

Please export your definitions and **attach** the file to your response. Use a `.txt` extension.

**Reply by cbcercas** [reply]
- all clusters have 2 nodes
- RabbitMQ 3.9.11
- Erlang 24.2
- docker images: rabbitmq:3-management-alpine and rabbitmq:3-management
Edit: Updated files below

**Reply by lukebakken** [reply]
Thank you. I should have also asked for the log files from all nodes in the cluster as well. If you can't provide the entire files, please provide log entries from the time period during the restart.

**Reply by cbcercas** [reply]
thank you, please found the logs in attachment

[definitions.rabbit_bunny-1_2022-1-12.txt](https://github.com/rabbitmq/rabbitmq-server/files/7854701/definitions.rabbit_bunny-1_2022-1-12.txt)
[bunny-1.log](https://github.com/rabbitmq/rabbitmq-server/files/7854697/bunny-1.log)
[bunny-2.log](https://github.com/rabbitmq/rabbitmq-server/files/7854698/bunny-2.log)
[project_1-1.log](https://github.com/rabbitmq/rabbitmq-server/files/7854699/project_1-1.log)
[project_1-2.log](https://github.com/rabbitmq/rabbitmq-server/files/7854700/project_1-2.log)

**Reply by lukebakken** [reply]
Thanks, I'll try to find time this week or next to reproduce this issue.

**Comment by ajmalrahman** [comment]
I faced the same on version: RabbitMQ 3.8.17 Erlang 23.3.4.2

The federation links disappear, re-creating them (the policies) shows the status again.

Steps to reproduce:

- Created the policy to federate the exchange on the downstream server
- Created a federation upstream on the downstream server
- Checked the status of the federation - the link status was Running (on node ha-1)
- Restarted all 3 nodes in the 3-node Rabbitmq cluster
- Checked the status of the federation, while restarting (ha-1) it went to Starting and then got removed.
- Created the policy again (manually) - the federation link came up again.

If it could be of any help, I have also attached the logs from all (3) nodes.

Logs:
[rmq-ha-2.txt](https://github.com/rabbitmq/rabbitmq-server/files/7856281/rmq-ha-2.txt)
[rmq-ha-0.txt](https://github.com/rabbitmq/rabbitmq-server/files/7856282/rmq-ha-0.txt)
[rmq-ha-1.txt](https://github.com/rabbitmq/rabbitmq-server/files/7856283/rmq-ha-1.txt)

(I didn't want to create a separate issue for the same thing)

**Reply by johanrhodin** [reply]
If you are on RabbitMQ 3.8.17 it could be this: https://github.com/rabbitmq/rabbitmq-server/issues/3148. Upgrade and try again.

**Reply by lukebakken** [reply]
@johanrhodin thanks for noticing that.

**Reply by ajmalrahman** [reply]
@johanrhodin I shall try upgrading then . So that means my issue could be different than the original issue posted here ?

**Reply by ajmalrahman** [reply]
I upgraded to the latest version (3.9.12) and it does not have this issue - the links do not disappear on a node restart. Thanks @lukebakken  and @johanrhodin

**Reply by lukebakken** [reply]
Thank you for letting us know!

**Comment by lukebakken** [comment]
I will try to find time over the next week or so to investigate.
